azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerTest.java

```diff
@@ -361,10 +362,20 @@ private void assertAttempts(String name, int attempt) {
   
   private ExecutableFlow prepareExecDir(File execDir, String flowName,
       int execId) throws IOException {
-    synchronized (this) {
-      // clean interrupted status
-      Thread.interrupted();
-      FileUtils.copyDirectory(execDir, workingDir);
+    // copyDirectory has been failing randomly with ClosedByInterruptException. have some retries.
```

Comment:
As long as the tests are stable and you are working on improving it, I will be happy to accpet this pull request after the conflicts are resolved.